```
>       _validate_search_traces(
            store, [exp_id], "name IN ('trace_0', 'trace_1')", trace_infos[:2][::-1]
        )

exp_id     = '359010433869461112'
generate_trace_infos = TraceInfos(trace_infos=[TraceInfo(request_id='298eba3a6dec48c9a22467b274ce3e63', experiment_id='359010433869461112', t...0052beae44adadd4b0ed6720b234', '9289e7fa5ad54e7aa5a934cc02caac36'], timestamps=[0, 10, 20, 30, 40, 50, 60, 70, 80, 90])
request_ids = ['298eba3a6dec48c9a22467b274ce3e63', '356eb0348fe848faafc8723c176a9ba4', '99799cfadf424fb990079397da218154', 'e6a0f6df477345[309](https://github.com/mlflow/mlflow/actions/runs/9318361069/job/25650537249?pr=12203#step:12:310)f5a50893df42007', 'a5916550982843c5b05f200a35945678', '1f76d43ecdf44585b8d4ac3459ff825b', ...]
store      = <mlflow.store.tracking.file_store.FileStore object at 0x7f1c7ab4c340>
timestamps = [0, 10, 20, 30, 40, 50, ...]
trace_infos = [TraceInfo(request_id='298eba3a6dec48c9a22467b274ce3e63', experiment_id='359010433869461112', timestamp_ms=0, executio...ytest-0/test_search_traces_filter0/mlruns/359010433869461112/traces/1f76d43ecdf44585b8d4ac3459ff825b/artifacts'}), ...]

...
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'mlflow.utils.search_utils.SearchTraceUtils'>, key_type = 'tag'
comparator = 'IN'

    @classmethod
    def is_tag(cls, key_type, comparator):
        if key_type == cls._TAG_IDENTIFIER:
            if comparator not in cls.VALID_TAG_COMPARATORS:
>               raise MlflowException(
                    f"Invalid comparator '{comparator}' not one of '{cls.VALID_TAG_COMPARATORS}"
                )
E               mlflow.exceptions.MlflowException: Invalid comparator 'IN' not one of '{'ILIKE', '!=', '=', 'LIKE'}

cls        = <class 'mlflow.utils.search_utils.SearchTraceUtils'>
comparator = 'IN'
key_type   = 'tag'
```

--- a/mlflow/utils/search_utils.py
+++ b/mlflow/utils/search_utils.py
@@ -1535,7 +1535,7 @@ class SearchTraceUtils(SearchUtils):
         """Filters a set of traces based on a search filter string."""
         if not filter_string:
             return traces
-        parsed = cls.parse_search_filter(filter_string)
+        parsed = cls.parse_search_filter_for_search_traces(filter_string)
 
         def trace_matches(trace):
             return all(cls._does_trace_match_clause(trace, s) for s in parsed)
@@ -1544,20 +1544,16 @@ class SearchTraceUtils(SearchUtils):
 
     @classmethod
     def _does_trace_match_clause(cls, trace, sed):
+        type_ = sed.get("type")
         key = sed.get("key")
         value = sed.get("value")
         comparator = sed.get("comparator").upper()
 
-        if key in cls.SEARCH_KEY_TO_TAG:
-            key = cls.SEARCH_KEY_TO_TAG[key]
+        if cls.is_tag(type_, comparator):
             lhs = trace.tags.get(key)
-        elif key in cls.SEARCH_KEY_TO_METADATA:
-            key = cls.SEARCH_KEY_TO_METADATA[key]
+        elif cls.is_request_metadata(type_, comparator):
             lhs = trace.request_metadata.get(key)
-        elif key in cls.SEARCH_KEY_TO_ATTRIBUTE:
-            key = cls.SEARCH_KEY_TO_ATTRIBUTE[key]
-            lhs = getattr(trace, key)
-        elif key in cls.VALID_SEARCH_ATTRIBUTE_KEYS:
+        elif cls.is_attribute(type_, key, comparator):
             lhs = getattr(trace, key)
         elif sed.get("type") == cls._TAG_IDENTIFIER:
             lhs = trace.tags.get(key)

--- a/tests/store/tracking/test_file_store.py
+++ b/tests/store/tracking/test_file_store.py
@@ -2930,12 +2930,6 @@ def test_search_traces_filter(generate_trace_infos):
     # filter by name
     _validate_search_traces(store, [exp_id], "name = 'trace_0'", trace_infos[:1])
     _validate_search_traces(store, [exp_id], "name != 'trace_0'", trace_infos[1:][::-1])
-    _validate_search_traces(
-        store, [exp_id], "name IN ('trace_0', 'trace_1')", trace_infos[:2][::-1]
-    )
-    _validate_search_traces(
-        store, [exp_id], "name NOT IN ('trace_0', 'trace_1')", trace_infos[2:][::-1]
-    )
     _validate_search_traces(store, [exp_id], "name LIKE 'trace_%'", trace_infos[::-1])
     _validate_search_traces(store, [exp_id], "name ILIKE 'Trace_%'", trace_infos[::-1])
     _validate_search_traces(
@@ -3031,8 +3025,6 @@ def test_search_traces_filter(generate_trace_infos):
         )
     _validate_search_traces(store, [exp_id], "run_id = 'run_5'", [trace_infos[5]])
     _validate_search_traces(store, [exp_id], "run_id != 'run_5'", trace_infos[6:][::-1])
-    _validate_search_traces(store, [exp_id], "run_id IN ('run_5')", [trace_infos[5]])
-    _validate_search_traces(store, [exp_id], "run_id NOT IN ('run_5')", trace_infos[6:][::-1])
     _validate_search_traces(store, [exp_id], "run_id LIKE 'run_%'", trace_infos[5:][::-1])
     _validate_search_traces(store, [exp_id], "run_id ILIKE 'RUN_5'", [trace_infos[5]])
 
@@ -3078,8 +3070,8 @@ def test_search_traces_filter(generate_trace_infos):
         ("foo.bar = 'baz'", r"Invalid entity type 'foo'"),
         ("invalid = 'foo'", r"Invalid attribute key 'invalid'"),
         ("trace.tags.foo = 'bar'", r"Invalid attribute key 'tags\.foo'"),
-        # TODO: This should raise
-        # ("trace.status < 'OK'", r"Invalid comparator '<'"),
+        ("trace.status < 'OK'", r"Invalid comparator '<'"),
+        ("name IN ('foo', 'bar')", r"Invalid comparator 'IN'"),
     ],
 )
 def test_search_traces_invalid_filter(generate_trace_infos, filter_string, error):

--- a/tests/store/tracking/test_file_store.py
+++ b/tests/store/tracking/test_file_store.py
@@ -3037,12 +3037,6 @@ def test_search_traces_filter(generate_trace_infos):
             store, [exp_id], f"{tag_identifier}.test_tag != 'tag_0'", trace_infos[1:][::-1]
         )
         _validate_search_traces(
-            store, [exp_id], f"{tag_identifier}.test_tag IN ('tag_0')", [trace_infos[0]]
-        )
-        _validate_search_traces(
-            store, [exp_id], f"{tag_identifier}.test_tag NOT IN ('tag_0')", trace_infos[1:][::-1]
-        )
-        _validate_search_traces(
             store, [exp_id], f"{tag_identifier}.test_tag LIKE 'tag_%'", trace_infos[::-1]
         )
         _validate_search_traces(

--- a/tests/store/tracking/test_sqlalchemy_store.py
+++ b/tests/store/tracking/test_sqlalchemy_store.py
@@ -4067,6 +4067,7 @@ def test_search_traces_with_filter(store_with_traces, filter_string, expected_id
         ("invalid = 'foo'", r"Invalid attribute key 'invalid'"),
         ("trace.tags.foo = 'bar'", r"Invalid attribute key 'tags\.foo'"),
         ("trace.status < 'OK'", r"Invalid comparator '<'"),
+        ("name IN ('foo', 'bar')", r"Invalid comparator 'IN'"),
     ],
 )
 def test_search_traces_with_invalid_filter(store_with_traces, filter_string, error):

